getter.go

```diff
@@ -152,26 +152,30 @@ func (g *getter) initChunks() {
 
 func (g *getter) worker() {
 	for c := range g.getCh {
-		g.retryGetChunk(c)
+		if err := g.retryGetChunk(c); err != nil {
+			close(g.readCh)
```

Comment:
If multiple workers have issues, this will make each of them close the channel potentially leading to a panic.

Reply:
Great catch, thanks. I'll have to think about that a bit more.

Reply:
How about wrapping it in a `sync.Once` like so:
```suggestion
			g.clOnce.Do(func() { close(g.readCh) })
```

Reply:
It's not super clear, but I think that would work in lieu of reorganising the flow of channels.